src/EnergyPlus/HVACDXSystem.cc

```diff
@@ -2180,186 +2198,154 @@ namespace HVACDXSystem {
-                                            ShowRecurringWarningErrorAtEnd(
-                                                DXCoolingSystem(DXSystemNum).DXCoolingSystemType + " \"" + DXCoolingSystem(DXSystemNum).Name +
-                                                    "\" - DX unit sensible part-load ratio calculation failed error continues. Sensible PLR "
-                                                    "statistics follow.",
-                                                DXCoolingSystem(DXSystemNum).DXCoilSensPLRFailIndex,
-                                                PartLoadFrac,
-                                                PartLoadFrac);
+                                } else if (SolFla == -2) {
+                                    PartLoadFrac = TempSpeedReqst / TempSpeedOut;
+                                    if (!WarmupFlag) {
+                                        if (DXCoolingSystem(DXSystemNum).DXCoilSensPLRFail < 1) {
+                                            ++DXCoolingSystem(DXSystemNum).DXCoilSensPLRFail;
+                                            ShowWarningError(DXCoolingSystem(DXSystemNum).DXCoolingSystemType +
+                                                             " - DX unit sensible part-load ratio calculation failed: part-load ratio limits "
+                                                             "exceeded, for unit = " +
+                                                             DXCoolingSystem(DXSystemNum).Name);
+                                            ShowContinueError("Estimated part-load ratio = " + RoundSigDigits(PartLoadFrac, 3));
+                                            ShowContinueErrorTimeStamp(
+                                                "The estimated part-load ratio will be used and the simulation continues. Occurrence info:");
                                         }
+                                        ShowRecurringWarningErrorAtEnd(DXCoolingSystem(DXSystemNum).DXCoolingSystemType + " \"" +
+                                                                           DXCoolingSystem(DXSystemNum).Name +
+                                                                           "\" - DX unit sensible part-load ratio calculation failed error "
+                                                                           "continues. Sensible PLR statistics follow.",
+                                                                       DXCoolingSystem(DXSystemNum).DXCoilSensPLRFailIndex,
+                                                                       PartLoadFrac,
+                                                                       PartLoadFrac);
                                     }
-                                } else {
+                                }
+                                CycRatio = PartLoadFrac;
+                            } else if ((TempSpeedOut - TempSpeedReqst) > Acc && NumOfSpeeds > 1) {
```

Comment:
If full load result at speed 1 does not exceed the set point and there are more speeds, then find the speed that exceeds the set point.